authors/apps/comments/views.py

```diff
@@ -0,0 +1,137 @@
+from ..articles.models import Articles
+from .serializers import (CommentsSerializer, CommentHistorySerializer)
+from rest_framework.permissions import IsAuthenticated
+from django.shortcuts import get_object_or_404
+
+
+class CreateCommentView(CreateAPIView, ListAPIView):
+    '''Endpoint for creating a comments'''
+    queryset = Comments.objects.all()
+    serializer_class = CommentsSerializer
+    lookup_field = 'article_slug'
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, article_slug):
+        '''Handles all post requests to create a comment'''
+
+        if Articles.objects.filter(slug=article_slug).exists():
+            # check if user is authorized
+            comment = request.data.get('comment', {})
+            article = Articles.objects.filter(slug=article_slug).first()
+            comment.update({'article': article})
+            comment.update({'user': request.user.id})
+
+            serializer = self.serializer_class(data=comment)
+            serializer.is_valid(raise_exception=True)
+            highlighted_text = comment.get('highlighted_text', None)
+            serializer.val_highlighted_text(highlighted_text, article)
+            comment['user'] = request.user
+            comment_ob = Comments(**comment)
+            comment_ob.save()
+            serializer = self.serializer_class(instance=comment_ob)
+            return Response(serializer.data,
+                            status=status.HTTP_201_CREATED)
+        else:
+            return Response({"error": "Article does not exist"},
+                            status=status.HTTP_404_NOT_FOUND)
+
+    def get(self, request, article_slug):
+        '''Handles all get requests by users to view the
+           comments of a particular article
+        '''
+        article = get_object_or_404(Articles, slug=article_slug)
+        queryset = self.get_queryset().filter(article=article)
+        serializer = CommentsSerializer(
+            queryset, many=True, context={'request': request})
+
+        count = len(serializer.data)
+
+        if count > 0:
+            return Response({"Comments": serializer.data,
+                             "Count": count},
+                            status=status.HTTP_200_OK)
+        else:
+            return Response({"error": "Article has no comments"},
+                            status=status.HTTP_204_NO_CONTENT)
+
+
+class UpdateDeleteCommentView(RetrieveUpdateDestroyAPIView):
+    '''Handles all requests for updating and deleting requests'''
+    queryset = Comments.objects.all()
+    serializer_class = CommentsSerializer
+    lookup_field = 'id'
+    permission_classes = (IsAuthenticated,)
+
+    def put(self, request, article_slug, id):
+        '''Handles all requests by user to update their comments'''
+        article = get_object_or_404(Articles, slug=article_slug)
+        comments = get_object_or_404(Comments, id=id, article=article)
+        if request.user.is_authenticated:
```

Comment:
Can you clarify why you have the `is_authenticated` check here?
 I think the permission class can handle the checking of whether the user is authenticated or not.

Reply:
Most of these are the remnants of my work after Martin changed it to an app. I will just delete the repetitive lines to fix the issues